src/Zenaton/v1/Model/Scheduling/Schedule.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Zenaton\Model\Scheduling;
+
+final class Schedule
+{
+    const API_DATE_FORMAT = 'Y-m-d\TH:i:s.u\Z';
+
+    /** @var string */
+    private $id;
+    /** @var string */
+    private $cron;
+    /** @var null|string */
+    private $name;
+    /** @var TaskTarget|WorkflowTarget */
+    private $target;
+    /** @var \DateTimeInterface */
+    private $insertedAt;
+    /** @var \DateTimeInterface */
+    private $updatedAt;
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * @internal should not be called by user code
+     *
+     * @return self
+     */
+    public static function fromArray(array $values = [])
+    {
+        $schedule = new static();
+        $schedule->id = (string) $values['id'];
+        $schedule->cron = (string) $values['cron'];
+        $schedule->name = isset($values['name']) ? (string) $values['name'] : null;
+        $schedule->target = static::createTarget($values['target']);
+        $schedule->insertedAt = \DateTimeImmutable::createFromFormat(static::API_DATE_FORMAT, $values['insertedAt']);
+        $schedule->updatedAt = \DateTimeImmutable::createFromFormat(static::API_DATE_FORMAT, $values['updatedAt']);
+
+        return $schedule;
+    }
+
+    /**
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCron()
+    {
+        return $this->cron;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return TaskTarget|WorkflowTarget
+     */
+    public function getTarget()
+    {
+        return $this->target;
+    }
+
+    /**
+     * @return \DateTimeInterface
+     */
+    public function getInsertedAt()
+    {
+        return $this->insertedAt;
+    }
+
+    /**
+     * @return \DateTimeInterface
+     */
+    public function getUpdatedAt()
+    {
+        return $this->updatedAt;
+    }
+
+    /**
+     * Creates a new instance of the schedule's target.
+     *
+     * @return TaskTarget|WorkflowTarget
+     */
+    private static function createTarget(array $targetData)
+    {
+        switch ($targetData['type']) {
```

Comment:
Having this case statement makes me think of ["Tell, don't ask"](https://www.martinfowler.com/bliki/TellDontAsk.html). I know we don't have abstract classes in the current php lib, but would it be possible to add this behaviour to the trait?

Reply:
I'm not sure to follow.

I'm well aware of the "Tell, don't ask" stuff, but for objects. Here we are only using a bare array. We cannot tell it anything, all we can do is ask I think.

I don't see what do you mean by adding this behavior to the trait. I don't see how the trait has anything to do with this code. The trait is only about convenience for our users. I should be able to do a `$client->scheduleWorkflow(...)` and have a complete Schedule object in return, even though I'm not using the trait.

I feel like this discussion is going to be quite difficult using comments, if you're available I'll gladly talk about that on a zoom / slack call.

Reply:
Sorry I thought this method was receiving an object, not an array. Don't mind me :see_no_evil: